File: server/feed3/src/main/java/info/glennengstrand/db/ParticipantDAO.java
package info.glennengstrand.db;

import org.skife.jdbi.v2.DBI;

import com.google.inject.Inject;

import info.glennengstrand.api.Participant;

public class ParticipantDAO extends RelationalDAO<Participant> {
	
	public Participant fetchParticipant(Long id) {
		return fetchSingle("call FetchParticipant(:id)", new ParticipantMapper(id), q -> q.bind("id", id));
	}
	
	public long upsertParticipant(String name) {
		return upsert("call UpsertParticipant(:moniker)", q -> q.bind("moniker", name));
	}

	@Inject
	public ParticipantDAO(DBI dbi) {
		super(dbi);
	}
	
}

File: server/feed11/src/main/scala/info/glennengstrand/news/dao/ParticipantDao.scala
package info.glennengstrand.news.dao

import scala.concurrent.ExecutionContext.Implicits.global
import slick.jdbc.MySQLProfile.api._
import scala.concurrent.Future
import info.glennengstrand.news.model.Participant

class ParticipantDao extends DataAccess[Participant] {
  override def fetchSingle(id: Int): Future[Participant] = {
    val s = sql"call FetchParticipant($id)".as[String]
    MySqlDao.db.run(s.map(r => Participant(Option(id), Option(r(0)), Option(toLink(id)))))
  }
  override def insert(p: Participant): Future[Participant] = {
    val s = sql"call UpsertParticipant(${p.name.get})".as[Int]
    MySqlDao.db.run(s.map(r => Participant(Option(r(0)), p.name, Option(toLink(r(0))))))
  }
}